homeassistant/components/iammeter/sensor.py

```diff
@@ -52,70 +60,68 @@ async def async_setup_platform(
     config_port = config[CONF_PORT]
     config_name = config[CONF_NAME]
     try:
-        async with async_timeout.timeout(PLATFORM_TIMEOUT):
-            api = await real_time_api(config_host, config_port)
-    except (IamMeterError, asyncio.TimeoutError) as err:
+        api = await hass.async_add_executor_job(
+            IamMeter, config_host, config_port, config_name
+        )
+    except asyncio.TimeoutError as err:
         _LOGGER.error("Device is not ready")
         raise PlatformNotReady from err
 
     async def async_update_data():
         try:
             async with async_timeout.timeout(PLATFORM_TIMEOUT):
-                return await api.get_data()
-        except (IamMeterError, asyncio.TimeoutError) as err:
+                return await hass.async_add_executor_job(api.client.get_data)
+        except asyncio.TimeoutError as err:
             raise UpdateFailed from err
 
     coordinator = DataUpdateCoordinator(
         hass,
         _LOGGER,
-        name=DEFAULT_DEVICE_NAME,
+        name=config_name,
         update_method=async_update_data,
         update_interval=SCAN_INTERVAL,
         request_refresh_debouncer=debounce.Debouncer(
             hass, _LOGGER, cooldown=0.3, immediate=True
         ),
     )
     await coordinator.async_refresh()
-    entities = []
-    for sensor_name, (row, idx, unit) in api.iammeter.sensor_map().items():
-        serial_number = api.iammeter.serial_number
-        uid = f"{serial_number}-{row}-{idx}"
-        entities.append(IamMeter(coordinator, uid, sensor_name, unit, config_name))
-    async_add_entities(entities)
-
-
-class IamMeter(CoordinatorEntity, SensorEntity):
-    """Class for a sensor."""
-
-    def __init__(self, coordinator, uid, sensor_name, unit, dev_name):
-        """Initialize an iammeter sensor."""
+    if coordinator.data["Model"] == DEVICE_3080:
+        async_add_entities(
+            IammeterSensor(coordinator, description)
+            for description in SENSOR_TYPES_3080
+        )
+    if coordinator.data["Model"] == DEVICE_3080T:
```

Comment:
If the above `if` is True, we don't have to test this one...

```suggestion
    elif coordinator.data["Model"] == DEVICE_3080T:
```